webapp/src/app/admin/teams/table/teams-table.component.ts

```diff
@@ -0,0 +1,241 @@
+import { lucideArrowUpDown, lucideChevronDown, lucideMoreHorizontal, lucideRotateCw, lucideXOctagon, lucidePlus, lucideCheck } from '@ng-icons/lucide';
+import { HlmButtonModule } from '@spartan-ng/ui-button-helm';
+import { HlmIconComponent, provideIcons } from '@spartan-ng/ui-icon-helm';
+import { HlmInputDirective } from '@spartan-ng/ui-input-helm';
+import { BrnMenuTriggerDirective } from '@spartan-ng/ui-menu-brain';
+import { HlmMenuModule } from '@spartan-ng/ui-menu-helm';
+import { BrnTableModule, PaginatorState } from '@spartan-ng/ui-table-brain';
+import { HlmTableModule } from '@spartan-ng/ui-table-helm';
+import { BrnSelectModule } from '@spartan-ng/ui-select-brain';
+import { HlmSelectModule } from '@spartan-ng/ui-select-helm';
+import { HlmSkeletonModule } from '@spartan-ng/ui-skeleton-helm';
+import { HlmCardModule } from '@spartan-ng/ui-card-helm';
+import { debounceTime, map } from 'rxjs';
+import { AdminService, TeamInfo } from '@app/core/modules/openapi';
+import { injectQueryClient } from '@tanstack/angular-query-experimental';
+import { octNoEntry } from '@ng-icons/octicons';
+import { HlmPopoverModule } from '@spartan-ng/ui-popover-helm';
+import { BrnPopoverComponent, BrnPopoverContentDirective, BrnPopoverTriggerDirective } from '@spartan-ng/ui-popover-brain';
+import { GithubLabelComponent } from '@app/ui/github-label/github-label.component';
+import { HlmScrollAreaComponent } from '@spartan-ng/ui-scrollarea-helm';
+
+const LOADING_TEAMS: TeamInfo[] = [
+  {
+    id: 1,
+    name: 'Team A',
+    color: '#FF0000',
+    repositories: [],
+    labels: []
+  },
+  {
+    id: 2,
+    name: 'Team B',
+    color: '#00FF00',
+    repositories: [],
+    labels: []
+  }
+];
+
+@Component({
+  selector: 'app-admin-teams-table',
+  standalone: true,
+  imports: [
+    FormsModule,
+    ReactiveFormsModule,
+
+    BrnMenuTriggerDirective,
+    HlmMenuModule,
+
+    BrnTableModule,
+    HlmTableModule,
+
+    HlmButtonModule,
+
+    HlmIconComponent,
+    HlmInputDirective,
+    HlmScrollAreaComponent,
+
+    BrnSelectModule,
+    HlmSelectModule,
+
+    HlmSkeletonModule,
+    HlmCardModule,
+
+    HlmPopoverModule,
+    BrnPopoverComponent,
+    BrnPopoverContentDirective,
+    BrnPopoverTriggerDirective,
+
+    GithubLabelComponent
+  ],
+  providers: [provideIcons({ lucideChevronDown, lucideMoreHorizontal, lucideArrowUpDown, lucideRotateCw, lucideXOctagon, lucidePlus, lucideCheck })],
+  templateUrl: './teams-table.component.html'
+})
+export class AdminTeamsTableComponent {
+  protected adminService = inject(AdminService);
+  protected queryClient = injectQueryClient();
+  protected octNoEntry = octNoEntry;
+
+  isLoading = input(false);
+  teamData = input.required<TeamInfo[] | undefined>();
+  allRepositories = input.required<Set<string> | undefined>();
+
+  _teams = computed(() => this.teamData() ?? LOADING_TEAMS);
+  protected readonly _rawFilterInput = signal('');
```

Comment:
Do we need all those `protected readonly`? To me it is clutter here. Also why use the `_` prefix?

Reply:
I've tried to keep the naming and attributes similar to Spartan's table example code: https://www.spartan.ng/components/data-table

It's not "necessary" per say, but it kind of provides a good indicator of what's "table-ui logic". If you dislike it, we can remove it of course though.